seqr/views/utils/pedigree_info_utils.py

```diff
@@ -425,120 +429,136 @@ def _parse_datstat_export_format(rows):
 
     return pedigree_rows
 
+def _get_detailed_yes_no(row, column, detail_column):
+    val = row[column]
+    formatted_val = val.title()
+    if val == DSMConstants.YES and row[detail_column]:
+        formatted_val = '{}; {}'.format(formatted_val, row[detail_column])
+    return formatted_val
+
+def _bool_condition_val(column_val, yes, no, default, unknown=None):
+    if column_val == DSMConstants.YES:
+        return yes
+    elif column_val == DSMConstants.NO:
+        return no
+    elif unknown and column_val == DSMConstants.UNSURE:
+        return unknown
+    return default
+
+def _test_summary(row, test):
+    test = test.strip()
+    if test == DSMConstants.OTHER:
+        display = 'Other tests: {}'.format(row[DSMConstants.OTHER_TEST_COLUMN] or 'Unspecified')
+    else:
+        display = DSMConstants.TEST_DISPLAYS[test]
 
-def _get_datstat_family_notes(row):
-    row = {k: v.encode('ascii', errors='ignore').decode() for k, v in row.items()}
+    if test not in DSMConstants.TEST_DETAIL_COLUMNS:
+        return display
 
-    DC = DatstatConstants
+    def _get_test_detail(column):
+        return row['TESTS_{}_{}'.format(test, column)]
 
-    def _get_column_val(column):
-        val_code = row[column].split(':')[0]
-        if column in DC.VALUE_MAP:
-            return DC.VALUE_MAP[column][val_code]
-        return val_code
+    relatives = _get_test_detail(DSMConstants.RELATIVES_KEY) or 'None Specified'
 
-    def _get_list_column_val(column):
-        return ', '.join([DC.VALUE_MAP[column][raw_val] for raw_val in row[column].split(':')[0].split(',')])
+    return '{name}. Year: {year}, Lab: {lab}, Relatives: {relatives}{other_relatives}'.format(
+        name=display,
+        year=_get_test_detail(DSMConstants.YEAR_KEY) or 'unspecified',
+        lab=_get_test_detail(DSMConstants.LAB_KEY) or 'unspecified',
+        relatives=', '.join([rel.strip().title().replace('_', ' or ') for rel in relatives.split(',')]),
+        other_relatives=': {}'.format(
+            _get_test_detail(DSMConstants.RELATIVE_DETAILS_KEY) or 'not specified') if DSMConstants.OTHER in relatives else '',
+    )
 
-    def _has_test(test):
-        return _get_column_val('TESTS.{}'.format(test)) == DC.YES
+def _get_testing(row):
+    tests = row[DSMConstants.TESTS_COLUMN]
+    if DSMConstants.NONE in tests or not tests:
+        return 'None'
+    elif DSMConstants.NOT_SURE_TEST in tests:
+        return 'Not sure'
+    return 'Yes;\n{tab}{tab}{tests}'.format(tab=DSMConstants.TAB, tests='\n{0}{0}'.format(DSMConstants.TAB).join([
```

Comment:
no because `DC.YES` is a constant that maps to the structured data we get from the DSM update - it could just as easily be `0` or `Y` as the actual string "Yes" at some point in the future, but in the outputted user-readable notes we always want to have the word "Yes"